sdk/eventhubs/azure-messaging-eventhubs/src/processor_partition_client.cpp

```diff
@@ -44,4 +69,34 @@ namespace Azure { namespace Messaging { namespace EventHubs {
 
     m_checkpointStore->UpdateCheckpoint(checkpoint, context);
   }
+
+  void ProcessorPartitionClient::UpdateCheckpoint(
+      Models::ReceivedEventData const& eventData,
+      Core::Context const& context)
+  {
+    uint64_t sequenceNumber{};
+    if (!eventData.SequenceNumber.HasValue())
+    {
+      throw std::runtime_error("Event does not have a sequence number.");
```

Comment:
It also may make sense to prefix the error message here, given that the customer is probably going to get this exeption after calling `sdkClient.doSomething()`.

Reply:
No other component does that in its exception messages, why is EventHubs special in that regard?

Reply:
I'm not looking at this in comparison, just thinking for the angle of diagnosability. If no other component does this, then I think they should. But Identity does :)
It is fine to not fix it in this PR if you want, but if you agree, please create an issue to address this later, maybe even as wide as for the whole SDK.